mobile/src/main/java/com/wedriveu/mobile/util/location/LocationService.java

```diff
@@ -0,0 +1,52 @@
+package com.wedriveu.mobile.util.location;
+
+import android.content.Intent;
+import android.location.Location;
+import com.google.android.gms.location.LocationRequest;
+
+/**
+ * Created by Nicola Lasagni on 19/07/2016.
+ */
+public interface LocationService {
+
+    String TAG = LocationService.class.getSimpleName();
+
+    int PERMISSION_REQUEST = 0;
+    int CHECK_SETTINGS_REQUEST = 1;
+    int LOCATION_REQUEST_INTERVAL = 10000;
+    int LOCATION_REQUEST_FASTEST_INTERVAL = 5000;
+    int LOCATION_REQUEST_PRIORITY = LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY;
+
+    Location getLastKnownLocation();
+
+    /**
+     * Checks for needed settings change.
+     */
+    void checkSettings();
+
+    /**
+     * Disables location and resets the last known position.
+     */
+    void disableLocationService();
+
+    /**
+     * This method must be called from the main Activity to handle Settings change.
+     * @param requestCode
```

Comment:
It was a suggestion-code, so since you've decided to use it for this task is your responsibility to make it clean, work well, and document it. 
Please, be professional.